src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java

```diff
@@ -13,15 +13,19 @@
     public void dateToString_ReturnsDateAsString() {
 
         // given
+        // assuming each element has different value, assertion can validate existence
+        // without worrying about order which depends of locale and timezone
         long date = 1506717341515L;
 
         // when
         String format = TimeUtils.dateToString(date);
 
         // then
         // validates only the date as the timezone and local can be different
-        assertThat(format).startsWith("2017-09-29");
-        // because of the timezone, hours depends whete the test is executed
+        assertThat(format).contains("2017");
+        assertThat(format).contains("09");
+        assertThat(format).contains("29");
```

Comment:
Don't see it fixing the test. When expected output was "2017-09-30 02:05:41 IST" 29 is not contained with that expected output :)

Reply:
Not sure what do you mean. I selected date so it makes sense and is not affected by the locals or timezone. Read assumptions above.

Reply:
Test is easily breakable.
Switching timezone from CEST works, to IST (Indian standard time)
Timezone can switch days too not only hours 👍 
```
java.lang.AssertionError: 
Expecting:
 <"2017-09-30 02:05:41 IST">
to contain:
 <"29"> 

	at jenkins.plugins.office365connector.TimeUtilsTest.dateToString_ReturnsDateAsString(TimeUtilsTest.java:27)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)```

Reply:
I would use a setup and teardown for this unit test.
In setup set default to UTC and in teardown set it back to default 👍

Reply:
```java
    static TimeZone defaultTimeZone;

    @BeforeClass
    public static void setup() {
        defaultTimeZone = TimeZone.getDefault();
        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
    }

    @AfterClass
    public static void tearDown() {
        TimeZone.setDefault(defaultTimeZone);
    }
```

Reply:
Documentation might help:

No reason for teardown I guess 👍 
```
/**
     * Sets the {@code TimeZone} that is returned by the {@code getDefault}
     * method. {@code zone} is cached. If {@code zone} is null, the cached
     * default {@code TimeZone} is cleared. This method doesn't change the value
     * of the {@code user.timezone} property.
     *
     * @param zone the new default {@code TimeZone}, or null
     * @throws SecurityException if the security manager's {@code checkPermission}
     *                           denies {@code PropertyPermission("user.timezone",
     *                           "write")}
     * @see #getDefault
     * @see PropertyPermission
     */
```